docs/_howto/typescript-swc.md

```diff
@@ -0,0 +1,257 @@
+
+#### Output of a SWC configured run of `npm test`
+
+```
+Other { toBeMocked: [Getter] }
+Other property descriptors {
+  __esModule: {
+    value: true,
+    writable: false,
+    enumerable: false,
+    configurable: false
+  },
+  toBeMocked: {
+    get: [Function: get],
+    set: undefined,
+    enumerable: true,
+    configurable: false
+  }
+}
+    1) should mock
+
+
+  0 passing (4ms)
+  1 failing
+
+  1) main
+       should mock:
+     TypeError: Descriptor for property toBeMocked is non-configurable and non-writable
+```
+
+#### Output of a `ts-node` configured run of `npm test`
+
+```
+Other { toBeMocked: [Function: toBeMocked] }
+Other property descriptors {
+  __esModule: {
+    value: true,
+    writable: false,
+    enumerable: false,
+    configurable: false
+  },
+  toBeMocked: {
+    value: [Function: toBeMocked],
+    writable: true,
+    enumerable: true,
+    configurable: true
+  }
+}
+mocked
+    ✔ should mock
+```
+
+The important difference to note about the object `Other` is that the property `toBeMocked` is a simple writable _value_ in the case of `ts-node` and a non-configurable _getter_ in the case of SWC. It being a getter is not a problem for Sinon, as we have a multitude of options for replacing those, but if `configurable` is set to `false` Sinon cannot really do anything about it.
+
+If we take a look at
+
+#### Conclusion of analysis
+
+SWC transforms the imports on the form `import * as Other from './other'` into objects where the individual exports are exposed through immutable accessors (_getters_).
+
+We can attack this issue in mainly 3 ways:
+
+1. somehow reconfigure SWC to produce different output when running tests that we can work with, either making writable values or configurable getters
+2. use pure dependency injection, opening up `./other.ts` to be changed from the inside
+3. attack how modules are loaded, injecting [an additional `require` "hook"](https://levelup.gitconnected.com/how-to-add-hooks-to-node-js-require-function-dee7acd12698)
+
+# Solutions
+
+## Mutating the output from the transpiler
+
+> [Working code][swc-mutable-export]
+
+If we can just flip the `configurable` flag to `true` during transpilation, Sinon could be instructed to replace the getter. Turns out, there is a SWC _plugin_ that does just that: [swc_mut_cjs_exports](https://www.npmjs.com/package/swc_mut_cjs_exports). By installing that and adding the following under the `jsc` key in `.swcrc`, you know get a configurable property descriptor.
+
+```json
+"experimental": {
+  "plugins": [[ "swc_mut_cjs_exports", {} ]]
+},
+```
+
+A getter _is_ different from a value, so you need to change your testcode slightly to replace the getter:
+
+```
+const stub = sandbox.fake.returns("mocked")
+sandbox.replaceGetter(Other, "toBeMocked", () => stub)
+```
+
+## Use pure dependency injection
+
+> [Working code][pure-di]
```

Comment:
Include link to section on testing in Single Page Apps In Depth

https://singlepageappbook.com/maintainability3.html#testing-interactions-between-modules